src/robotFactory.js

```diff
@@ -1,10 +1,87 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x
+        ? coords.x
+        : 0,
+      y: coords.y
+        ? coords.y
+        : 0,
```

Comment:
We can write it shorter.
```suggestion
      x: coords.x || 0,
      y: coords.y || 0,
```